src/core/packet.lua

```diff
@@ -102,6 +102,7 @@ function length (p) return p.length end
 -- Set packet data length.
 function resize (p, len)
    assert(len <= max_payload, "packet payload overflow")
+   ffi.fill(p.data + p.length, math.max(0, len - p.length))
```

Comment:
This was indeed tested on lugano-1 (Intel(R) Xeon(R) CPU E5-1650 v3 @ 3.50GHz). I think the most interesting part is actually the (annotated) profiler output:

```
$ sudo lock ./snabb snabbmark esp 1e7 1024 encapsulate Fpv
locking /var/lock/lab.. flock: getting lock took 0.000002 seconds
81%  lib/ipsec/aes_128_gcm.lua:encrypt
  <- 100%  TRACE  24          ->loop     
 4%  lib/ipsec/esp.lua:encapsulate
  <- 92%  TRACE  24          ->loop     
  <-  8%  JIT Compiler
 3%  program/snabbmark/snabbmark.lua:esp
  <- 100%  Garbage Collector
Encapsulation (packet size = 1024): 13.15 Gbit/s
$ sudo lock ./snabb snabbmark esp 1e7 1024 decapsulate Fpv
locking /var/lock/lab.. flock: getting lock took 0.000002 seconds
83%  lib/ipsec/aes_128_gcm.lua:decrypt
  <- 100%  TRACE   7          ->loop     
 5%  lib/ipsec/esp.lua:decapsulate
  <- 100%  TRACE   7          ->loop     
 3%  program/snabbmark/snabbmark.lua:esp
  <- 94%  Garbage Collector
  <-  6%  TRACE   7          ->loop     
 3%  core/packet.lua:clone
  <- 100%  TRACE   7          ->loop     
Decapsulation (packet size = 1024): 13.86 Gbit/s
```

We see 80% of the time used by our AES DynASM routines (“a” for annotate shows this more definitively). I read this as 20% overhead due to ESP on 1KB packets (vs plain AES-GCM).